File: 525-contiguous-array/contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
        HashMap<Integer,Integer> array=new HashMap<>();
        array.put(0,-1);
        int sum=0;
        int max=0;
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i]==0)
                sum--;
            else
                sum++;
            if(array.containsKey(sum))
            {
                int a=i-array.get(sum);
                if(a>max)
                max=a;
            }
            else
            array.put(sum,i);
        }
        return max;
    }
}

File: 525-contiguous-array/contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        s=dict()
        s[0]=-1
        max=0
        sum=0
        for i in range(len(nums)):
            if(nums[i]==0):
                sum=sum-1
            else:
                sum=sum+1
            ele=s.get(sum,-2)
            if(ele==-2):
                s[sum]=i
            else:
                a=i-s[sum]
                if(a>max):
                    max=a
        return max